src/test/model/graph.test.ts

```diff
@@ -141,9 +141,78 @@ describe('Graph', () => {
 
     describe('invariants', () => {
 
-        it('removes extra properties not supported by node definition');
-        it('ensures the correct order properties');
-        it('initializes all properties with default values');
+        it('keeps only firstmost properties supported by node definition', async () => {
+            const loader = new GraphLoader();
+            const graph = await loader.loadGraph({
+                nodes: [
+                    {
+                        id: 'res',
+                        ref: 'add',
+                        props: [
+                            { key: 'c', value: '10' },
+                            { key: 'a', value: '12' },
+                            { key: 'b', value: '21' },
+                        ]
+                    },
+                    {
+                        id: 'p',
+                        ref: 'string',
+                        props: [
+                            { key: 'value', value: '42' },
+                            { key: 'value', value: '68' },
+                        ]
+                    }
+                ],
+                refs: {
+                    add: runtime.defs['math.add'],
+                    string: runtime.defs['string'],
+                }
+            });
+            assert.strictEqual(Object.keys(graph.nodes).length, 2);
+            const node1 = graph.getNodeById('res');
+            const node2 = graph.getNodeById('p');
+            assert.strictEqual(node1?.props.length, 2);
+            assert.strictEqual(node2?.props.length, 1);
+            assert.strictEqual(Number(node1?.props[0].value) + Number(node1?.props[1].value), 33);
```

Comment:
This assertion needs to be a bit more explicit; I sort of understand the point, but the fact that the `math.add`'s implementation computes a sum of two numbers isn't relevant to what you're trying to test here. So just assert the two property values separately.